### PR TITLE
Chainable Globally Attachable Listener Manager

### PR DESCRIPTION
**Pull request by Vampire**
This contains a merge of my other PR and the changes for generation for the DiscordApiBuilder.
I didn't include getters for now as I doubt how useful and properly implementable they are,
especially regarding suppliers and functions.

### CODE REVIEW

File: gradle/listener-manager-generation.gradle [hunk 1]
```diff
@@ -568,6 +642,121 @@ def addGenericGloballyAttachableListenerMethods(attachableListenerManagerInterfa
             .addParameter('GloballyAttachableListener', 'listener')
             .removeBody()
 
+    chainableGloballyAttachableListenerManagerInterface
+            .addMethod('addListener')
+            .setJavadocComment('''
+                Adds a {@code GloballyAttachableListener} to all created {@code DiscordApi} instances.
+                Adding a listener multiple times will only add it once
```
Comment (Saladoc) [inline gradle/listener-manager-generation.gradle]: Returning `ListenerManager`s should not be referred here.

**Review by Saladoc — CHANGES_REQUESTED** [review]
This looks to be exactly what's needed.

I've been trying to think of a way to hand out the `ListenerManager`s via a `BiConsumer<DiscordApi,ListenerManager>` callback but that seems to be overkill for such a use case. After all, global listeners are seldom cancelled.

File: gradle/listener-manager-generation.gradle (hunk 1 above)
Comment (Vampire) [reply gradle/listener-manager-generation.gradle]: You are totally right of course, copy & paste error fixed.

**Comment by Vampire** [comment]
I think the `BiConsumer<DiscordApi,ListenerManager>` can independently considered when it is also considered for all other methods to be able to easily remove a listener from inside a lambda in a separate PR.

**Pull request merged**